Gopkg.toml

```diff
@@ -41,3 +41,7 @@
 [[constraint]]
   name = "github.com/openfaas/faas-provider"
   version = "0.14.0"
+
+[[constraint]]
+  name = "github.com/docker/cli"
```

Comment:
When I looked at the containerd resolve it seemed cleaner / more lightweight without depending on Docker itself, like this - https://github.com/containerd/containerd/blob/master/cmd/ctr/commands/resolver.go

Do you think it can be made to work without the Docker CLI?

Reply:
like 	"github.com/containerd/containerd/remotes/docker" using NewDockerAuthorizer?

Even it (https://github.com/containerd/containerd/blob/4f348c5b5b53a70b282ab5b81b2cc8f03bf6a338/remotes/docker/authorizer.go#L90) didn't seem to use Docker itself.

What do you think?

Reply:
We can't easily remove dependency on docker/cli, mostly because I believe we want to support cred helper binaries, though I think I've heard that there is an alternative go pkg for that.

Reply:
Maybe we should open docker/cli PR to reduce dependencies

Reply:
Is this needed to support the helper binaries then? Does containerd/cri not support pulling from AWS ECR then?

Reply:
https://medium.com/@xynova/keeping-aws-registry-pull-credentials-fresh-in-kubernetes-2d123f581ca6

Reply:
I wonder what @ibuildthecloud did in k3c? https://github.com/rancher/k3c/blob/master/pkg/daemon/image.go

I can see a reference to docker/distribution there actually